lib/dockerregistry/inventory.go

```diff
@@ -587,6 +589,60 @@ func ToPQIN(registryName RegistryName, imageName ImageName, tag Tag) string {
 	return string(registryName) + "/" + string(imageName) + ":" + string(tag)
 }
 
+func getImagesSliceFromMap(images map[ImageName]DigestTags) []Image {
+	var result []Image
+
+	for name, dmap := range images {
+		result = append(result, Image{
+			ImageName: name,
+			Dmap:      dmap,
+		})
+	}
+
+	return result
+}
+
+// getImagesFromDestRegistry returns all images from not src repository
+func (sc *SyncContext) getImagesFromDestRegistry(
```

Comment:
Can we simplify this to only read out 1 registry, specified from the CLI as an argument for `--gen-images` ? There is no need to discern "src" or "dest" registries. You would just call ReadRepository() against some registry, which would populate sc.Inv for us for that 1 registry. Then you can use the getImageSliceFromMap() conversion function you wrote to generate the `images:` field.

A better flag name might be just `--snapshot`.

Reply:
To clarify, we could indeed read out multiple registries and take their intersection (`--gen-images` could take a list of registries instead of just 1), but I think in actual practice it would probably not be used. The use-case we are trying to support is snapshotting `gcr.io/google-containers`.